offline-coverage validation & ocamlbackend cvt-bool-bv

ailrst opened the pull request:
- Change offline OCaml backend's implementation of cvt-bool-bv to just emit the function as we support this in BASIL but not ite expressions. 
- Add the rASL checks to offline-coverage
- Make coverage test rules explicitly depend on the test binary so that updating the coverage binary invalidates diffs properly


The offline lifter doesn't match the load structure or intrinsic restriction of the offline. The load structure isn't a problem downstream but it would be good to get rid of eq_enum. 

eq_enum appears in [cbz/cbnz opcodes](https://head.aslp-web-staging.pages.dev/?opcode=0x35ffffe0&bytes=E0+FF+FF+35&asm=cbnz+w0%2C+%230xfffffffffffffffc&debug=0). Its now fixed by performing this simplification in the IBI. Without looking into it too closely, I'm assuming there is some opcode dependency that prevents the [symbolic simplification](https://github.com/UQ-PAC/aslp/blob/partial_eval/libASL/symbolic.ml#L774) but there's probably a smarter solution available there.

ailrst committed:
offline output cvt_bool_bv tapply instead of ite

ailrst committed:
add explicit asloff-cov dependency on offlinecoverage rule

ailrst committed:
run rasl checks on offline coverage

ailrst committed:
version 0.3.1

ailrst committed:
add explicit asli dependency for coverage rules

katrinafyi commented on CHANGES.md:
```diff
@@ -1,4 +1,10 @@
 
+
+0.3.1 (2025-02-04)
```
date?

katrinafyi commented on tests/coverage/dune:
```diff
@@ -32,12 +32,12 @@
+(rule (deps ../../bin/offline_coverage.exe) (action (with-stdout-to off_cov_float   (run ./off_run "aarch64_float_.+"))))
+(rule (deps ../../bin/offline_coverage.exe) (action (with-stdout-to off_cov_vector1 (run ./off_run "aarch64_vector_arithmetic_binary.+"))))
+(rule (deps ../../bin/offline_coverage.exe) (action (with-stdout-to off_cov_vector2 (run ./off_run "aarch64_vector_\\(arithmetic_unary\\|[^a]\\).+"))))
+(rule (deps ../../bin/offline_coverage.exe) (action (with-stdout-to off_cov_memory  (run ./off_run "aarch64_memory_.+"))))
```
i don't like this but i know you've tried very hard to fix this and it has to be this way. thanks

ailrst committed:
simplify eq_enum in offline_utils

ncough committed:
Inject simplications into offline transform

* Identify where conditional lifttime simplifications can be added
to the offline transform.

* Add a transform to eliminate eq_enum from the ARM spec. Only
possibe as there are no uses of eq_enum between runtime values.

* Remove eq_enum from offline utils
